pkg_security.Rmd

```diff
@@ -1,19 +1,73 @@
+    
+    * Do not print the API key even in verbose mode.
+
+* In the GitHub issue template, it should be stated not to share any credentials (it's the case of [rOpenSci issue template](https://docs.ropensci.org/rodev/reference/use_ro_github.html)). If your user accidentally shares credentials in an issue, make sure they're aware of that so they can't revoke the key.
+
+### Secrets in packages and development
+
+You'll need to protect your secrets as you protect your users' secrets, but there's more to take into account and keep in mind.
+
+#### Secrets and recorded requests in tests
+
+If you use [`vcr`](https://docs.ropensci.org/vcr/) or [`httptest`](https://enpiar.com/r/httptest/) in tests for caching API responses, you need to make sure the recorded requests / fixtures do not contain secrets. Refer to [`vcr` guidance](https://books.ropensci.org/http-testing/security.html) and [`httptest` guidance "Redacting and Modifying Recorded Requests"](https://enpiar.com/r/httptest/articles/redacting.html), and inspect your recorded requests / fixtures before committing them the first time to be sure you got the setup right.
+
+`vcr` being an rOpenSci package, you can post any question you might have to [rOpenSci forum](https://discuss.ropensci.org/)
+
+#### Share secrets with CI services
 
+Now, you might need to share secrets with CI services such as Travis or GitHub Actions.
+
+* You could store API keys as environment variables / secrets, referring to the docs of the CI service.
+
+* For bigger secrets like tokens, refer [to the googlesheets guidance](https://cran.r-project.org/web/packages/googlesheets/vignettes/managing-auth-tokens.html#the-making-of-googlesheets.tar.gz-for-cran) that involves in particular the encryption of tokens for sharing them with Travis. Document the steps you made in a Markdown file so you, or say a new maintainer, can remember how to do that next time.
+
+#### Secrets and collaborations
+
+What about pull requests from external contributors? Tests using your secrets will fail, so you might want to skip them depending on the context. For instance, in your CI account you could create an environment variable called `THIS_IS_ME` and then skip tests based on the presence of this variable. This obviously means the PR checks by the CI are not exhaustive, so you'll need to check out the PR locally to run all tests.
+
+You could also choose [a more complicated setup](https://blog.algolia.com/travis-encrypted-variables-external-contributions/).
+
+Document the behavior of your package for external PRs in [CONTRIBUTING.md](#friendlyfiles) for the sake of people making PRs and of people reviewing them (you in a few weeks, and other authors of the package).
+
+### Secrets and CRAN
+
+Skip any tests (`skip_on_cran()`) and examples (`dontrun`) requiring credentials on CRAN.
```

Comment:
might add to the end of this sentence, for example: 

> , even if the test request/response is cached with vcr.

Reply:
Oh, why would one skip the tests when the response is cached?

Reply:
I suppose a reason for skipping tests on CRAN is to make sure the package checks just because the Suggested dependency `vcr` isn't installed... https://www.stats.ox.ac.uk/pub/bdr/noSuggests/ was it what you had in mind @sckott?

Reply:
if a key/etc. is required for the test to run, even if cached, it wouldn't work on cran,

Reply:
aaah ok I see cf https://github.com/ropensci/vcr/issues/137

Reply:
right

Reply:
do you have any idea whether it's the same with httptest?

Reply:
no, don't know